pkg/gofr/cmd/terminal/progress.go

```diff
@@ -0,0 +1,105 @@
+package terminal
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+)
+
+type ProgressBar struct {
+	stream  Output
+	current int64
+	total   int64
+	tWidth  int
+	mu      sync.Mutex
+}
+
+type Term interface {
+	IsTerminal(fd int) bool
+	GetSize(fd int) (width, height int, err error)
+}
+
+func NewProgressBar(out Output, total int64) *ProgressBar {
+	w, _, err := out.getSize()
+	if err != nil {
+		fmt.Printf("error getting terminal size, err : %v, could not initialize progress bar\n", err)
```

Comment:
Why not printing to output?

But also, why not simply returning the error , and let the caller handles it?

Reply:
Returning error to let users figure out if the progress bar was initialized! 
Initial thoughts were to not handle any error as we would like the process to continue even if the progress bar is not initialized as it is just a visualization effect.